spec.html

```diff
@@ -30767,7 +30767,7 @@ <h1>The RegExp Constructor</h1>
         <li>is <dfn>%RegExp%</dfn>.</li>
         <li>is the initial value of the *"RegExp"* property of the global object.</li>
         <li>creates and initializes a new RegExp object when called as a function rather than as a constructor. Thus the function call `RegExp(&hellip;)` is equivalent to the object creation expression `new RegExp(&hellip;)` with the same arguments.</li>
-        <li>is designed to be subclassable. It may be used as the value of an `extends` clause of a class definition. Subclass constructors that intend to inherit the specified RegExp behaviour must include a `super` call to the RegExp constructor to create and initialize subclass instances with the necessary internal slots.</li>
+        <li>may be used as the value of an `extends` clause of a class definition. Subclass constructors that intend to inherit the specified RegExp behaviour must include a `super` call to the RegExp constructor to create and initialize subclass instances with the necessary internal slots.</li>
```

Comment:
sadly, RegExp was explicitly designed _to_ be subclassable

Reply:
I'm still curious. Does anyone use subclassable RegExp in wild? I never seen that

Reply:
Neither have i, but the committee decided not to remove that plumbing, and it was definitely designed for it.

Reply:
> sadly, RegExp was explicitly designed _to_ be subclassable

Doesn't mean we need to mention that in the specification, though.

Reply:
I think it’s reasonable to denote the things that are built for it, to separate them from the ones that aren’t.

Reply:
I disagree that we should explicitly mark things that are built for it, for the reason stated above: that we're nudging future proposals. Yes, it was explicitly designed for it in the past, a past that we regret.

Reply:
> I'm still curious. Does anyone use subclassable RegExp in wild? I never seen that

[At least Babel.](https://babeljs.io/repl#?browsers=ie%2011&build=&builtIns=false&corejs=3.6&spec=false&loose=false&code_lz=G4QwTgBGCmELwQPQAoD8AeAntcA-AOgCYDeALAL4CUAtGugLYD2AdgC4AWBJATFbRoRCYuxXpUQBzANxA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=true&targets=&version=7.14.4&externalPlugins=)

Reply:
From reading that output, it seems like Babel is doing that primarily so that it interoperates properly just in case someone else *is* using subclassable regexps - i'd bet they could drastically simplify their output if they didn't have to worry about all the regexp symbols.

Reply:
@ljharb could you show an example that will properly work with `.exec`, `.match`, `.replace` without it?

Reply:
If “properly work” means “RegExp subclasses don’t work, which is fine because nobody actually uses them except for core-js feature detection”, then they’d just use those methods directly. If by “properly work” you mean that RegExp subclasses continue to work, then no, of course those are required.

Reply:
@ljharb by "properly" work I mean how to make work extended transpiled functionality of `RegExp` - NCG in this case work with `String.prototype.match` and `String.prototype.replace`. It's absolutely not related to polyfills - it's the implementation of ES2018 feature on ES2015.

Reply:
For that, you’re probably right - but “making transpilers possible” isn’t a primary concern when designing features.

Reply:
However, it's a significant plus.

Reply:
When possible, yes, but it's not worth the complexity and performance hits that the RegExp system takes. Unfortunately, we can't remove it because there's no practical way to tease apart polyfill feature detections from real code when gathering usage data.